pkg/ddl/ingest/checkpoint.go

```diff
@@ -34,52 +34,57 @@ import (
 	"go.uber.org/zap"
 )
 
-// CheckpointManager is a checkpoint manager implementation that used by non-distributed reorganization.
+// CheckpointManager is a checkpoint manager implementation that used by
+// non-distributed reorganization. It manages the data as two-level checkpoints:
+// "flush"ed to local storage and "import"ed to TiKV. The checkpoint is saved in
+// a table in the TiDB cluster.
 type CheckpointManager struct {
-	ctx       context.Context
-	flushCtrl FlushController
-	sessPool  *sess.Pool
-	jobID     int64
-	indexIDs  []int64
+	ctx           context.Context
+	cancel        context.CancelFunc
+	flushCtrl     FlushController
+	sessPool      *sess.Pool
+	jobID         int64
+	indexIDs      []int64
+	localStoreDir string
+	logger        *zap.Logger
 
 	// Derived and unchanged after the initialization.
 	instanceAddr     string
 	localDataIsValid bool
 
 	// Live in memory.
-	checkpoints     map[int]*TaskCheckpoint // task ID -> checkpoint
-	mu              sync.Mutex
-	minTaskIDSynced int
-	dirty           bool
+	mu          sync.Mutex
+	checkpoints map[int]*taskCheckpoint // task ID -> checkpoint
+	// we require each task ID to be continuous and start from 1.
+	minTaskIDFinished int
+	dirty             bool
 	// Local meta.
-	pidLocal   int64
-	startLocal kv.Key
-	endLocal   kv.Key
+	pidFlushed      int64
+	startKeyFlushed kv.Key
+	endKeyFlushed   kv.Key
 
 	// Persisted to the storage.
-	minKeySyncLocal  kv.Key
-	minKeySyncGlobal kv.Key
-	localCnt         int
-	globalCnt        int
+	minKeyFlushed  kv.Key
+	minKeyImported kv.Key
```

Comment:
shouldn't we call it maxKeyImported, it means all keys < it are imported

Reply:
ChatGPT still prefer the minXXX

---

You
How to name a variable which means "all keys less than it are flushed, but the keys larger than it can be flushed or not flushed". And it's calculated from the minimum value of some concurrency worker's progress.

ChatGPT
You might consider naming the variable something descriptive yet succinct, like "minFlushedKey" or "flushThreshold". This would indicate that it represents the minimum key value up to which all keys are flushed, but keys larger than it might or might not be flushed depending on their status. Alternatively, you could name it "minFlushedProgressKey" to be more explicit about its relationship to progress.

Here's a breakdown of each potential name:

minFlushedKey: This name clearly indicates that it represents the minimum key value that has been flushed. It suggests that any keys smaller than this value are flushed, while those larger than it may or may not be flushed.
flushThreshold: This name implies that it's a point or level that determines when flushing occurs. It's like a threshold that, once reached, triggers the flushing of keys. "FlushThreshold" conveys that keys below this threshold are flushed, while those above it may or may not be.
minFlushedProgressKey: This name explicitly links the variable to progress and flushing. It suggests that it represents the minimum key based on the progress made, and keys less than or equal to this value are considered flushed.
Choose the name that best fits the context and naming conventions of your project. The most important thing is that the name is clear and understandable to anyone reading your code.

Reply:
i don't think `minFlushedKey` suggests that any keys smaller than this value are flushed, it only states that it's the min key that have flushed and any key before or after it may or may not flushed

> And it's calculated from the minimum value of some concurrency worker's progress.

from this description, we can only got a name like min key of on-going task

Reply:
How about rename it to `flushedKeyLowWatermark`? I can do it in https://github.com/pingcap/tidb/pull/52993

Reply:
lgtm